### PR TITLE
fix: handle zero blockhash in reverted log

### PR DESCRIPTION
**Pull request by makramkd**
Ticket: https://smartcontract-it.atlassian.net/browse/VRF-741

Buggy RPCs can sometimes send us zero blockhashes in the reverted log and we end up not processing it correctly. The only fix seems to be to resubscribe and replay, so that's what we do in this change.

### CODE REVIEW

**Review by jmank88 — DISMISSED** [review]

File: core/chains/evm/log/broadcaster.go [hunk 1]
```diff
@@ -519,17 +529,22 @@ func (b *broadcaster) invalidatePool() int64 {
...
+func (b *broadcaster) onNewLog(log types.Log) (shouldResubscribe bool) {
 	b.maybeWarnOnLargeBlockNumberDifference(int64(log.BlockNumber))
 
 	if log.Removed {
 		// Remove the whole block that contained this log.
 		b.logger.Debugw("Found reverted log", "log", log)
 		b.logPool.removeBlock(log.BlockHash, log.BlockNumber)
-		return
+		// this can happen with buggy RPC's, the only fix is to replay from the block before the reverted log
+		if (log.BlockHash == common.Hash{}) || log.BlockNumber == 0 {
```
Comment (reductionista) [inline core/chains/evm/log/broadcaster.go:540]: Ideally, we should add a test for this case

Comment (makramkd) [reply core/chains/evm/log/broadcaster.go:540]: Agreed, looking at how to add a test.

**Comment by makramkd** [comment]
Added a bunch of unit tests that should cover the lines added.

**Comment by makramkd** [comment]
Superseded by #11174

**Pull request closed without merging**